src/classes/USER_UserService_TEST.cls

```diff
@@ -0,0 +1,247 @@
+        System.assertEquals(true, retrievedErrorTypeAfterEmailSent[0].Email_Sent__c);
+    }
+
+
+    /*********************************************************************************************************
+    * @description Tests logging and Error__c and sending and email to System Administrators.
+    * @return void
+    **********************************************************************************************************/
+    private static testMethod void testNotifySysAdminsAboutInActiveIsFrozenSysAdminUser() {
+        Test.startTest();
+        USER_UserService.notifySysAdminsAboutInActiveIsFrozenSysAdminUser();
+        Test.stopTest();
+
+        List<Error__c> retrievedErrorTypeAfterEmailSent = [SELECT Email_Sent__c FROM Error__c WHERE Object_Type__c = :USER_UserService.OBJECT_TYPE_USER];
+        System.assertNotEquals(null, retrievedErrorTypeAfterEmailSent);
+    }
+
+
+    /*********************************************************************************************************
+    * @description Tests logging and Error__c record.
+    * @return void
+    **********************************************************************************************************/
+    private static testMethod void testLogInActiveIsFrozenUserError() {
+        Test.startTest();
+        USER_UserService.logInActiveIsFrozenUserError();
+        Test.stopTest();
+
+        List<Error__c> retrievedErrorType = [SELECT Full_Message__c FROM Error__c WHERE Object_Type__c = :USER_UserService.OBJECT_TYPE_USER];
+        System.assertEquals(Label.InactiveFrozenScheduledJobsOwnerErrMsg, retrievedErrorType[0].Full_Message__c);
+    }
+
+
+    /*********************************************************************************************************
+    * @description Tests the sending of the Scheduled Jobs error email to System Administrators.
+    * @return void
+    **********************************************************************************************************/
+    private static testMethod void testSendInActiveFrozenUserEmail() {
+        Test.startTest();
+        USER_UserService.logInActiveIsFrozenUserError();
+
+        USER_UserService.sendInActiveIsFrozenUserEmail();
+        Test.stopTest();
+
+        List<Error__c> retrievedErrorTypeAfterEmailSent = [SELECT Email_Sent__c FROM Error__c WHERE Object_Type__c = :USER_UserService.OBJECT_TYPE_USER];
+        System.assertEquals(true, retrievedErrorTypeAfterEmailSent[0].Email_Sent__c);
+    }
+
+
+    /*********************************************************************************************************
+    * @description Utility method to create Users with System Administrator profile and sets their IsActive status accordingly.
+    * @return List containing the newly created Users.
+    **********************************************************************************************************/
+    private static List<User> createUsers() {
```

Comment:
rather than duplicate this helper method, why not make the one in USER_FrozenInActiveUser_TDTM_TEST public and use it, or vice-versa.

Reply:
Didn't know if we wanted to keep a clear delineation between data creation for test classes. However, making the createUsers method public is a good because we can later move it and other setup code into a class that handles test data setup like the UTIL_UnitTestData_TEST. I didn't do this initially to keep the User functionality encapsulated.